2d_classification/MIL/mil_classify_monai.py

```diff
@@ -0,0 +1,944 @@
+
+            for i in range(images.shape[0]):
+
+                img = images[i]
+                if np.random.random() > self.p: img = np.flip(img, axis=1)
+                if np.random.random() > self.p: img = np.flip(img, axis=2)
+                if np.random.random() > self.p: img = np.transpose(img, axes=(0,2,1))
+                images2.append(img)
+
+            d[key] = np.stack(images2, axis=0)
+
+        return d
+
+
+def main():
+
+    args = parse_args()
+
+    if args.distributed:
+        ngpus_per_node = torch.cuda.device_count()
+        args.optim_lr = ngpus_per_node * args.optim_lr / 2 #heuristic to scale up learning rate in multigpu setup
+        args.world_size = ngpus_per_node * args.world_size
+
+        print('Multi-gpu', ngpus_per_node, 'rescaled lr', args.optim_lr)
+        mp.spawn(main_worker, nprocs=ngpus_per_node, args=(args,))
+    else:
+        # Simply call main_worker function
+        main_worker(0, args)
+
+def main_worker(gpu, args):
+
+    args.gpu = gpu
+    num_classes = 5
+    max_tiles = 44 # max tiles in inference at a time
+    tile_size=256
+
+
+    if args.distributed:
+        args.rank = args.rank * torch.cuda.device_count() + gpu
+        dist.init_process_group(backend=args.dist_backend, init_method=args.dist_url, world_size=args.world_size, rank=args.rank)
+
+    print(args.rank, ' gpu', args.gpu)
+
+    torch.cuda.set_device(args.gpu) #use this default device (same as args.device if not distributed)
+    torch.backends.cudnn.benchmark = True
+
+    if args.rank==0:
+        print('Batch size is:', args.batch_size, 'epochs', args.epochs)
+
+    #############
+    data_dir = '/PandaChallenge2020/train_images/'
+
+    with open('panda_training_fold0.json') as json_file:
+        training_list = json.load(json_file)
+        for d in training_list:
+            d['image'] = os.path.join(data_dir, d['image'])
+
+    with open('panda_validation_fold0.json') as json_file:
+        validation_list = json.load(json_file)
+        for d in validation_list:
+            d['image'] = os.path.join(data_dir, d['image'])
+
+
+    train_transform = transforms.Compose(
+        [
+            LoadTiffd(keys=["image"]), #replace with LoadImageD
+            # transforms.LoadImageD(keys=["image"], reader='ITKReader', dtype=np.uint8), #not working, can't specify TIFF page number (need page=1), and it outputs 4 channels not 3 for some reason
+            LabelEncodeIntegerGraded(keys=["label"], num_classes=num_classes), #encode label
+            # transforms.Lambdad(keys=["label"], func=lambda x: (np.arange(num_classes)<int(x)).astype(np.float32)), #encode label (same but can't pickle for multiprocessing in dataloader)
+            ImageExtractPatchesd(keys=["image"], tile_count=max_tiles, tile_size=tile_size, tile_random_offset=True, tile_all=False, tile_background_val=255),
+            BatchedRandFlip(keys=["image"]),
+            transforms.CastToTypeD(keys=["image"], dtype=np.float32),
```

Comment:
I submitted a PR for the `dtype` of ScaleIntensityRange: https://github.com/Project-MONAI/MONAI/pull/3235.

Thanks.

Reply:
Thanks